### ISSUE TITLE
Missing module imports don't cause error

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#23961]
cc @andy-ms 

**TypeScript Version:**  master

**Search Terms:** 

**Code**

```ts
import './404';
```

**Expected behavior:**

Can't find module error

**Actual behavior:**

pass

**Playground Link:**  

**Related Issues:**

**Comment by andy-ms** [comment]
This has come up before -- `import "foo";` is unsafe and isn't checked. At some point in the past we didn't check it and now we have to do the same for backwards-compatibility reasons. Duplicate of #20534

**Comment by falsandtru** [comment]
I see: #20576
